hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/index/bloom/SparkHoodieMetadataBloomIndexHelper.java

```diff
@@ -0,0 +1,167 @@
+            final List<Pair<String, IndexFileFilter>> result = new ArrayList<>();
+            final List<Pair<String, String>> columnStatKeys = new ArrayList<>();
+            long partitionCount = 0;
+            while (partitionItr.hasNext()) {
+              String partitionName = partitionItr.next();
+              List<String> partitionFileNameList = HoodieIndexUtils.getLatestBaseFilesForPartition(partitionName,
+                      hoodieTable).stream().map(baseFile -> baseFile.getFileName())
+                  .collect(toList());
+              partitionCount++;
+
+              if (partitionFileNameList.isEmpty()) {
+                result.add(new ImmutablePair<String, IndexFileFilter>(partitionName,
+                    new IntervalTreeBasedIndexFileFilter(partitionName, Collections.emptyList())));
+                continue;
+              }
+
+              for (String fileName : partitionFileNameList) {
+                Pair<String, String> partitionFileNameKey = Pair.of(partitionName, fileName);
+                columnStatKeys.add(partitionFileNameKey);
+              }
+            }
+            LOG.debug("Loaded base files for " + partitionCount + " partitons, time taken: " + timer.endTimer());
+
+            try {
+              timer = new HoodieTimer().startTimer();
+              Map<Pair<String, String>, HoodieMetadataColumnStats> fileToColumnStatMap = hoodieTable
+                  .getMetadataTable().getColumnStats(columnStatKeys, keyField);
+              LOG.debug("Loaded column ranges for " + columnStatKeys.size() + " keys, time taken: " + timer.endTimer());
+
+              Map<String, List<BloomIndexFileInfo>> partitionToFilesBloomInfoMap = fileToColumnStatMap.entrySet()
+                  .stream().map(entry -> {
+                    final BloomIndexFileInfo bloomIndexFileInfo = new BloomIndexFileInfo(
+                        FSUtils.getFileId(entry.getKey().getRight()),
+                        entry.getValue().getMinValue(),
+                        entry.getValue().getMaxValue());
+                    return new ImmutablePair<>(entry.getKey().getLeft(), bloomIndexFileInfo);
+                  }).collect(Collectors.groupingBy(p -> p.getKey(), Collectors.mapping(p -> p.getValue(), toList())));
+
+              result.addAll(partitionToFilesBloomInfoMap.entrySet().stream().map(entry -> {
+                return new ImmutablePair<String, IndexFileFilter>(entry.getKey(),
+                    new IntervalTreeBasedIndexFileFilter(entry.getKey(), entry.getValue()));
+              }).collect(Collectors.toList()));
+            } catch (MetadataNotFoundException me) {
+              throw new HoodieMetadataException("Unable to find column range metadata!", me);
+            }
+            return result.iterator();
+          }
+        }, true);
+    return HoodieJavaRDD.of(partitionIndexFileFilterRDD);
+  }
+
+  @Override
+  public HoodiePairData<HoodieKey, HoodieRecordLocation> findMatchingFilesForRecordKeys(
```

Comment:
if I understand correctly, there is no difference in this method compared to whats in master (1st dag for metadata bloom index) ?